ant-cli/src/actions/download.rs

```diff
@@ -21,19 +24,25 @@ use color_eyre::{
 use std::path::PathBuf;
 
 pub async fn download(addr: &str, dest_path: &str, client: &Client) -> Result<()> {
-    let public_address = str_to_addr(addr).ok();
-    let private_address = crate::user_data::get_local_private_archive_access(addr)
-        .inspect_err(|e| error!("Failed to get private archive access: {e}"))
-        .ok();
-
-    match (public_address, private_address) {
-        (Some(public_address), _) => download_public(addr, public_address, dest_path, client).await,
-        (_, Some(private_address)) => download_private(addr, private_address, dest_path, client).await,
-        _ => Err(eyre!("Failed to parse data address {addr}"))
+    let try_public_address = str_to_addr(addr).ok();
+    if let Some(public_address) = try_public_address {
+        return download_public(addr, public_address, dest_path, client).await;
+    }
+
+    let try_private_address = crate::user_data::get_local_private_archive_access(addr).ok();
+    if let Some(private_address) = try_private_address {
+        return download_private(addr, private_address, dest_path, client).await;
+    }
+
+    let try_datamap = DataMapChunk::from_hex(addr).ok();
+    if let Some(datamap) = try_datamap {
+        return download_from_datamap(addr, datamap, dest_path, client).await;
+    }
+
+    Err(eyre!("Failed to parse data address {addr}"))
             .with_suggestion(|| "Public addresses look like this: 0037cfa13eae4393841cbc00c3a33cade0f98b8c1f20826e5c51f8269e7b09d7")
             .with_suggestion(|| "Private addresses look like this: 1358645341480028172")
-            .with_suggestion(|| "Try the `file list` command to get addresses you have access to"),
```

Comment:
lack of suggestion of `Data_map_chunk address look like`  ?

Reply:
Added a suggestion mentioning them